service/task.go

```diff
@@ -640,6 +641,81 @@ func (uis *UIServer) taskLogRaw(w http.ResponseWriter, r *http.Request) {
 	uis.render.Stream(w, http.StatusOK, data, "base", "task_log.html")
 }
 
+func (uis *UIServer) taskFileRaw(w http.ResponseWriter, r *http.Request) {
+	projCtx := MustHaveProjectContext(r)
+	if projCtx.Task == nil {
+		http.Error(w, "task not found", http.StatusNotFound)
+		return
+	}
+
+	fileName := gimlet.GetVars(r)["file_name"]
+	if fileName == "" {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New("file name not specified"))
+		return
+	}
+
+	taskFiles, err := artifact.GetAllArtifacts([]artifact.TaskIDAndExecution{{TaskID: projCtx.Task.Id, Execution: projCtx.Task.Execution}})
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to find artifacts for task: %s", projCtx.Task.Id))
+		return
+	}
+	taskFiles, err = artifact.StripHiddenFiles(taskFiles, true)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to strip hidden files for task: %s", projCtx.Task.Id))
+		return
+	}
+	var tFile *artifact.File
+	for _, taskFile := range taskFiles {
+		if taskFile.Name == fileName {
+			tFile = &taskFile
+			break
+		}
+	}
+	if tFile == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.New("File not found"))
+		return
+	}
+	if !utility.StringMatchesAnyRegex(tFile.ContentType, uis.Settings.Ui.FileStreamingContentTypes) {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New(fmt.Sprintf("unsupported file content type: %s", tFile.ContentType)))
+		return
+	}
+
+	response, err := http.Get(tFile.Link)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "downloading file:"))
+		return
+
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		uis.LoggedError(w, r, response.StatusCode, errors.Errorf("failed to download file with status code: %d", response.StatusCode))
+		return
+	}
+
+	// Create a buffer to stream the file in chunks
+	bufferSize := 1000000 // 1mb
+	buffer := make([]byte, bufferSize)
+
+	w.Header().Set("Content-Type", tFile.ContentType)
+	// Stream the file content to the response writer
+	for {
+		n, err := response.Body.Read(buffer)
+		if err == io.EOF {
+			break // End of file
```

Comment:
nit: this is slightly easier to read as a return

Reply:
I think the `End of file` comment is unnecessary since the if statement is already checking for an `io.EOF` error.